ompi/runtime/ompi_spc.h

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2018      The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ *
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#ifndef OMPI_SPC
+#define OMPI_SPC
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <dlfcn.h>
+
+#include "ompi/communicator/communicator.h"
+#include "ompi/datatype/ompi_datatype.h"
+#include "ompi/runtime/params.h"
+#include "opal/mca/timer/timer.h"
+#include "opal/mca/base/mca_base_pvar.h"
+#include "opal/util/argv.h"
+#include "opal/util/show_help.h"
+#include "opal/util/output.h"
+
+#include MCA_timer_IMPLEMENTATION_HEADER
+
+/* INSTRUCTIONS FOR ADDING COUNTERS
+ * 1.) Add a new counter name in the ompi_spc_counters_t enum before
+ *     OMPI_SPC_NUM_COUNTERS below.
+ * 2.) Add corresponding counter name and descriptions to the
+ *     counter_names and counter_descriptions arrays in
+ *     ompi_spc.c  NOTE: The names and descriptions
+ *     MUST be in the same array location as where you added the
+ *     counter name in step 1.
+ * 3.) If this counter is based on a timer, add its enum name to
+ *     the logic for timer-based counters in the ompi_spc_init
+ *     function in ompi_spc.c
+ * 4.) Instrument the Open MPI code base where it makes sense for
+ *     your counter to be modified using the SPC_RECORD macro.
+ *     Note: If your counter is timer-based you should use the
+ *     SPC_TIMER_START and SPC_TIMER_STOP macros to record
+ *     the time in cycles to then be converted to microseconds later
+ *     in the ompi_spc_get_count function when requested by MPI_T
+ */
+
+/* This enumeration serves as event ids for the various events */
+enum ompi_spc_counters_t {
```

Comment:
My suggestion earlier was in shorthand.  This should probably be:

`typedef enum ompi_spc_counters { ... } ompi_spec_counters_t`

I think this will make it like many other declarations in OMPI.